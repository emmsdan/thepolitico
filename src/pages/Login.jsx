import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import CallToAction from '../components/presentationals/CallToAction/CallToAction';
// import PropTypes from 'prop-types'
import FormCard from '../components/presentationals/Form/FormCard';
import FormInput from '../components/presentationals/Form/FormInput';
// import FormButton from '../components/presentationals/Form/FormButton';
import { generateFormData } from '../utils/form';
import { authenticationRequestHandler } from '../store/reducers/auth';
import Loader from '../components/presentationals/Loader/Loader';
import { authenticationToken } from '../utils/helpers';

class Login extends Component {
  componentDidMount() {
    const { history } = this.props;
    authenticationToken() && history.push('/');
  }

  submitLoginDetails = event => {
    event.preventDefault();
    const formData = generateFormData(event.target);
    const {
      history,
      location: { state },
    } = this.props;
    let redirectUrl = state ? state.from.pathname : 'account';
    return this.props.authenticationRequestHandler(
      formData,
      'login',
      history,
      redirectUrl,
    );
  };

  render() {
    return (
      <>
        {this.props.auth.isLoading && (
          <Loader loader="show" text="Hi, am processing your input" />
        )}
        <CallToAction withButton={false} />
        <section className="wrapper">
          <FormCard
            header="Sign in to Continue"
            onSubmit={this.submitLoginDetails}
          >
            <FormInput
              name="username"
              id="login_username"
              placeholder="enter email or phone number"
              title="Email or Phone"
            />

            <FormInput
              name="password"
              id="login_password"
              placeholder="enter password"
              title="Password"
              type="password"
            />
            <br />
            <div className="field">
              <button type="submit" className="button-bigger">
                Login
              </button>
            </div>
            <br />
            <a href="forgot-password"> Forgot My Password </a>
            <br />
            <br />
            <a href="register"> Create a new Account </a>
            <br />
            <br />
          </FormCard>
        </section>
      </>
    );
  }
}

Login.propTypes = {
  auth: PropTypes.object,
  authenticationRequestHandler: PropTypes.func.isRequired,
  location: PropTypes.object,
  history: PropTypes.object,
};

const mapStateToProps = state => {
  return { auth: state.auth };
};

export default connect(
  mapStateToProps,
  { authenticationRequestHandler },
)(Login);
