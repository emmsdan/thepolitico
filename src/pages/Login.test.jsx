import React from 'react';
import { mount } from 'enzyme';

import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import Login from './Login';
let store = {};
describe('Login', () => {
  beforeEach(() => {
    const mockStore = configureMockStore();
    store = mockStore({
      history: [],
      location: {},
      auth: { isLoading: true },
      authenticationRequestHandler: jest.fn(),
    });
  });

  it('should render correctly', () => {
    const component = mount(<Login store={store} />);
    expect(component).toMatchSnapshot();
  });

  it('should simulate submit correctly', () => {
    const component = mount(
      <Provider store={store}>
        <Login />
      </Provider>,
    );
    const node = component.find('button').simulate('click');
    expect(node).toMatchSnapshot();
  });
});
