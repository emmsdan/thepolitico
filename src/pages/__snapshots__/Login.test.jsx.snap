// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
<ConnectFunction
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Login
    auth={
      Object {
        "isLoading": true,
      }
    }
    authenticationRequestHandler={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Loader
      loader="show"
      text="Hi, am processing your input"
    >
      <div
        className="center load-overlay show"
      >
        <div>
          <div
            className="loader center"
            style={
              Object {
                "backgroundImage": "url([object Object])",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "contain",
                "borderRadius": 0,
              }
            }
          />
          <strong>
            Hi, am processing your input
            ...
          </strong>
        </div>
      </div>
    </Loader>
    <CallToAction
      withButton={false}
    >
      <section
        className="call-to-action"
      >
        <div
          className="placeholder"
        >
          <h2>
            Every Vote Count
          </h2>
          <h4>
            Vote and Be Voted for
          </h4>
        </div>
      </section>
    </CallToAction>
    <section
      className="wrapper"
    >
      <FormCard
        header="Sign in to Continue"
        onSubmit={[Function]}
      >
        <form
          className="form-fields"
          header="Sign in to Continue"
          onSubmit={[Function]}
        >
          <fieldset
            className="fieldset"
          >
            <legend>
              Sign in to Continue
            </legend>
            <FormInput
              id="login_username"
              name="username"
              placeholder="enter email or phone number"
              title="Email or Phone"
            >
              <div
                className="field"
              >
                <span
                  className="error red-text"
                  htmlFor="login_username"
                />
                <input
                  id="login_username"
                  name="username"
                  placeholder="enter email or phone number"
                  title="Email or Phone"
                />
                <label
                  htmlFor="login_username"
                >
                  Email or Phone
                  :
                </label>
              </div>
            </FormInput>
            <FormInput
              id="login_password"
              name="password"
              placeholder="enter password"
              title="Password"
              type="password"
            >
              <div
                className="field"
              >
                <span
                  className="error red-text"
                  htmlFor="login_password"
                />
                <input
                  id="login_password"
                  name="password"
                  placeholder="enter password"
                  title="Password"
                  type="password"
                />
                <label
                  htmlFor="login_password"
                >
                  Password
                  :
                </label>
              </div>
            </FormInput>
            <br />
            <div
              className="field"
            >
              <button
                className="button-bigger"
                type="submit"
              >
                Login
              </button>
            </div>
            <br />
            <a
              href="forgot-password"
            >
               Forgot My Password 
            </a>
            <br />
            <br />
            <a
              href="register"
            >
               Create a new Account 
            </a>
            <br />
            <br />
          </fieldset>
        </form>
      </FormCard>
    </section>
  </Login>
</ConnectFunction>
`;

exports[`Login should simulate submit correctly 1`] = `
<button
  className="button-bigger"
  type="submit"
>
  Login
</button>
`;
